apps/weather/src/common/filters/grpc-exception.filter.ts

```diff
@@ -0,0 +1,23 @@
+import { Catch, Logger, RpcExceptionFilter } from '@nestjs/common';
+import { RpcException } from '@nestjs/microservices';
+import { throwError } from 'rxjs';
+import { status } from '@grpc/grpc-js';
+import { Errors } from '../constants/errors.const';
+
+@Catch()
+export class GrpcExceptionFilter implements RpcExceptionFilter {
+  catch(exception: Error) {
+    if (exception instanceof RpcException) {
+      return throwError(() => exception.getError());
+    }
+
+    Logger.error(exception.message);
+
+    return throwError(() =>
+      new RpcException({
+        code: status.INTERNAL,
+        message: Errors.INTERNAL_SERVER_ERROR,
+      }).getError()
+    );
+  }
+}
```

Comment:
You can create a `shared` library within your monorepo to store all reusable logic in one place.